### ISSUE TITLE
Clear credentials woth aws cli configure

### ISSUE THREAD

**Issue by ouadie-lahdioui** [aws/aws-cli#3346]
Hello everybody 👋 

I juste want to know if it's possible to remove a named profile credentials with ```aws cli configure```.

Thank you in advanced

**Comment by kyleknap** [comment]
Seems reasonable. Out of curiosity, how would you plan to use this feature? An example would be great. Otherwise, marking as feature request.

**Comment by cyrilchampier** [comment]
We do use this workaround to "simulate" a profile deletion:
`aws configure --profile PROFILE_NAME set aws_secret_access_key ''`

**Comment by rbeede** [comment]
This would be a useful feature to have.  I accidentally mistyped a profile name and ended up creating one by mistake.

`aws configure set region us-west-1 --profile PROFILE_NAME_oops_typo`

I'd like to not see that oops_typo with I  list-profiles  later.   The command should be consistent with the naming conventions used elsewhere:

`aws configure delete-profile --profile PROFILE_NAME_oops_typo`
Make the --profile option a REQUIRED parameter to avoid deleting the default profile by mistake.  If someone wants to delete the default profile they should use `--profile default`

**Comment by murphysm2342** [comment]
> This would be a useful feature to have. I accidentally mistyped a profile name and ended up creating one by mistake.
> 
> `aws configure set region us-west-1 --profile PROFILE_NAME_oops_typo`
> 
> I'd like to not see that oops_typo with I list-profiles later. The command should be consistent with the naming conventions used elsewhere:
> 
> `aws configure delete-profile --profile PROFILE_NAME_oops_typo`
> Make the --profile option a REQUIRED parameter to avoid deleting the default profile by mistake. If someone wants to delete the default profile they should use `--profile default`

Would be great if this was available.  I've just run into an issue where setting values to "" throws an error of either missing source_profile or mismatch if I put in a soure_profile.  Just want to purge the profile altogether so I can re-create it with the correct config.  Luckily I'm early in our stages of development so won't be a major impact to users if I just purge their config file altogether.

**Comment by mohammedimdad** [comment]
Not at all working

**Comment by KenBerg75** [comment]
Our use case for this would be on the build server. We are using Terraform with multiple aws providers so using env vars is not an option. We generate the credentials file, and run our build/deploy. It would be nice to remove the file after completion, so we do not leave secrets on the build server.

**Comment by rehanvdm** [comment]
Same with MFA, setting it to "" is not enough, you have to remove the line completely

**Comment by YevheniiPokhvalii** [comment]
Any news on this? I have a script that can create bulk of MFA profiles. Also, users may have profiles in other than default places. It would be great to have a CLI command that can remove a profile with its credentials safely.

**Comment by YevheniiPokhvalii** [comment]
> This feature would be useful to have

For now, I've created a script to address some lack of aws cli features. Well, it may not be ideal, probably I'll need to refactor it one day, but it is robust and works as expected for me. You can try if you wish. It should be sourced, because it exports AWS variables in the current shell. Flag -d removes a profile from .aws files.
https://github.com/YevheniiPokhvalii/generate-aws-tokens

**Comment by tb3088** [comment]
`sed  '/^\[profile <PROFILE_TO_DEL>/,/^\[/ { /^\[profile <PROFILE_TO_DEL>/ d; /^\[/ !d; }' AWS_CONFIG_FILE`
this will stomp on any comments or lines you might have between [profile] stanzas. A trivial change will run against the CREDENTIALS_FILE

I've got a pile of wrappers around AWS CLI. And one of them on my TODO is wrapping 'aws configure' so you can do bloody obvious taks like this. Feel free to poke around [my repo.](https://github.com/tb3088/shell-environment/blob/master/.bashrc_aws)

**Comment by JimBledsoe** [comment]
I would definitely +1 this request. You can add and list them through the CLI, why not be able to delete them without having to manually edit sensitive files.

**Comment by beporter** [comment]
> I would definitely +1 this request. You can add and list them through the CLI, why not be able to delete them without having to manually edit sensitive files.

Agreed. Anything you can do via `aws configure ...`, you should be able to **undo** via `aws configure ...`. 

It's probably accomplishable with some creative use of `sed` on the config files directly, but that's not always portable. It should come built into the awscli to begin with.
